employee-rostering-frontend/src/ui/pages/availability/AvailabilityEvent.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import EmployeeAvailability from "domain/EmployeeAvailability";
+import { useTranslation } from "react-i18next";
+import { Split, SplitItem, Button, Level, LevelItem } from "@patternfly/react-core";
+import { TrashIcon, OkIcon, WarningTriangleIcon, ErrorCircleOIcon } from '@patternfly/react-icons';
+
+export interface AvailabilityEventProps {
+  availability: EmployeeAvailability;
+  updateEmployeeAvailability: (ea: EmployeeAvailability) => void;
+  removeEmployeeAvailability: (ea: EmployeeAvailability) => void;
+}
+  
+const AvailabilityEvent: React.FC<AvailabilityEventProps> = (props: AvailabilityEventProps) => {
+  const { t } = useTranslation();
+  return (
+    <span
+      data-tip
+      data-for={String(props.availability.id)}
+      className="availability-event"
+  
+    >
+      <Split>
+        <SplitItem isFilled={false}>{t("EmployeeAvailabilityState." + props.availability.state)}</SplitItem>
+        <SplitItem isFilled />
+        <SplitItem isFilled={false}>
+          <Button
+            onClick={() => props.removeEmployeeAvailability(props.availability)}
+            variant="danger"
+          >
+            <TrashIcon />
+          </Button>
+        </SplitItem>
+      </Split>
+      <Level gutter="sm">
+        <LevelItem>
+          <Button
+            onClick={() => props.updateEmployeeAvailability({
+              ...props.availability,
+              state: "DESIRED"
+            })}
+            style={{
```

Comment:
Buttons are too big and in most displays the unavailable one don't appear. Maybe them 10px and the margin 1px?
In addition, for smaller unavailable slots (e.g. 1-hour) the buttons don't appear at all which makes it impossible for users to modify/delete this unavailability.
Add an [onSelectEvent](http://intljusticemission.github.io/react-big-calendar/examples/index.html#prop-onSelectEvent) that shows details of the event and these buttons or the EditAvailabilityModal (see how it works on google calendar).

Reply:
Cannot shrink button much; already at min-content. Add popup (a bit wonky, but I will fix it in a later PR where I refactor the how the Roster Pages work).

Reply:
You can add `fontSize: "10px"` for example in the `style` to make the buttons smaller (min size of button depends on the size of the text inside it). See what font size looks ideal and displays all buttons in medium screen sizes.

Reply:
And yeah the pop-up only shows for larger events because it's a nested component inside the event so it's after the text, when event is small you can't even click on it (same problem). But it's fine to be fixed in a later PR.